### PR TITLE
xref issues x 3

### PR DESCRIPTION
**Pull request by dlrice**
As several issues were codependent these have all been worked on in this branch:

https://www.ebi.ac.uk/panda/jira/browse/TRM-23604
* Logic to decide when and how to the implicit databases

--

https://www.ebi.ac.uk/panda/jira/browse/TRM-23140
* EMBL, GenBank, DDBJ, not only EMBL
* ENST, ENSG and ENSP not just ENST

--

https://www.ebi.ac.uk/panda/jira/browse/TRM-23381
* Miscellaneous databases and Protocols and Materials databases sections

### CODE REVIEW

File: src/view/uniprotkb/components/XRefView.tsx [hunk 1]
```diff
@@ -11,52 +11,53 @@ import {
...
-  databases: XrefsGoupedByDatabase[];
-  primaryAccession: string;
-};
+import {
+  DatabaseInfoPoint,
+  AttributesItem,
+} from '../../../model/types/DatabaseTypes';
+import { sortBy } from '../../../utils/utils';
 
 type XrefItem = {
```
Comment (xwatkins) [inline src/view/uniprotkb/components/XRefView.tsx]: Looks like this is not used

File: src/view/uniprotkb/components/XRefView.tsx [hunk 2]
```diff
@@ -65,97 +66,218 @@ export const getPropertyString = (property: Property) => {
...
     <Fragment>
-      <XRefExternalLink url={info.uriLink} accession={primaryAccession} id={id}>
-        {id}
-      </XRefExternalLink>{' '}
-      {properties} {isoformLink}
+      {linkNode} {propertiesNode} {isoformNode}
     </Fragment>
   );
 };
+type XRefExternalLinkProps = {
```
Comment (xwatkins) [inline src/view/uniprotkb/components/XRefView.tsx]: I think this is not used?

File: src/view/uniprotkb/components/XRefView.tsx [hunk 3]
```diff
@@ -11,52 +11,53 @@ import {
...
-}): JSX.Element => {
-  let link = url;
-  if (accession) {
-    link = link.replace(/%acc/g, accession);
-  }
-  if (id) {
-    link = link.replace(/%value/g, id);
-  }
-  return <ExternalLink url={link}>{children}</ExternalLink>;
+export const transfromProperties = (properties: Property[]) => {
```
Comment (xwatkins) [inline src/view/uniprotkb/components/XRefView.tsx]: Wonder if this could be a model util?

File: src/view/uniprotkb/components/XRefView.tsx [hunk 4]
```diff
@@ -11,52 +11,53 @@ import {
...
   databaseCategory: string;
   xref: Xref;
 };
 
-type XRefExternalLinkProps = {
-  url: string;
-  accession?: string | null | undefined;
-  id?: string | null | undefined;
-  children: string | string[];
+export const fillUrl = (
```
Comment (xwatkins) [inline src/view/uniprotkb/components/XRefView.tsx]: Wonder if something like `createExternalLinkUrl` would be a better name?

File: src/data/databaseInfo.json [hunk 5]
```diff
@@ -1,1514 +1,1589 @@
...
+	},
+	{
+		"name": "RefSeq",
+		"displayName": "RefSeq",
+		"category": "SEQ",
+		"uriLink": "https://www.ncbi.nlm.nih.gov/protein/%id",
+		"attributes": [{
+			"name": "NucleotideSequenceId",
+			"xmlTag": "nucleotide sequence ID",
+			"uriLink": "https://www.ncbi.nlm.nih.gov/nuccore/%necleotideId"
```
[the remark this answers is not in the document]
Comment (dlrice) [reply src/data/databaseInfo.json]: Fixed

File: webpack.common.js [hunk 6]
```diff
@@ -13,9 +13,11 @@ module.exports = {
       'react-router-dom': path.resolve('./node_modules/react-router-dom'),
     },
   },
+  node: {
+    fs: 'empty'
```
[the remark this answers is not in the document]
Comment (dlrice) [reply webpack.common.js]: I honestly don't know. Removed it.

File: src/view/uniprotkb/components/XRefView.tsx (hunk 2 above)
Comment (dlrice) [reply src/view/uniprotkb/components/XRefView.tsx]: Good catch

File: src/view/uniprotkb/components/XRefView.tsx (hunk 1 above)
Comment (dlrice) [reply src/view/uniprotkb/components/XRefView.tsx]: Another good catch

File: src/view/uniprotkb/components/XRefView.tsx (hunk 3 above)
Comment (dlrice) [reply src/view/uniprotkb/components/XRefView.tsx]: That'a a good call

**Pull request merged**